### ISSUE TITLE
warning: this clause cannot match because of different types/sizes

### ISSUE THREAD

**Issue by bill-mybiz** [CrowdHailer/OK#18]
When adding some doctests for the OK lib this morning, I ran `mix test` and got a bunch of warnings:

```
warning: this clause cannot match because of different types/sizes
```

I checked, and my code base also had these warnings. It happens anytime that you return a literal or an `OK.success` expression as the last line.

I was going to just post this issue, but then I started looking into it and I think I've actually addressed it! I personally dislike PRs without a corresponding issue, so I'm creating this issue first. (If you prefer just the PR, then in the future I'll just do that.)

The issue stems from the following clause of `nest/1` (I added :

```
  defp nest([normal | []]) do
    quote do
      case unquote(normal) do
        {:ok, value} ->
          {:ok, value}
        {:error, reason} ->
          {:error, reason}
      end
    end
  end
```

The warning stems from the `case` statement not having a match on the `{:error, reason}` when the line is already an `{:ok, value}` statement (or the corresponding `OK.success` equivalent). So in the PR I'm about to submit, I added a couple clauses that specifically pattern match out these two cases (ok literal and `OK.success`).

**Comment by CrowdHailer** [comment]
Don't mind having an issue with each PR. I have added a comment in the PR, with the opinion that I don't think lots of special cases is the right way to handle this problem.

I do quite like keeping PR's on a single issue. I would like to look quite a bit more at these code changes however the errata/docs changes in the PR is great and I would merge them asap

**Comment by CrowdHailer** [comment]
Merged the PR, Thanks for the contributions @bill-mybiz I think you had probably thought it through further than I had managed to, reread your rational about each of the cases to consider and decided you were talking sense. Does that make this issue resolved? Also there is a 1.5.1 release on hex with the changes

**Comment by bill-mybiz** [comment]
LGTM :+1: :tada:
